app/models/user.rb

```diff
@@ -398,15 +398,21 @@ def log_failed_login
   end
 
   def log_successful_login
-    update_attribute(:last_login_on, Time.now)
+    update_attribute(:last_login_on, Time.current)
   end
 
   def pref
     preference || build_preference
   end
 
   def time_zone
-    @time_zone ||= (pref.time_zone.blank? ? nil : ActiveSupport::TimeZone[pref.time_zone])
+    @time_zone ||= ActiveSupport::TimeZone[pref.time_zone] || ActiveSupport::TimeZone["Etc/UTC"]
```

Comment:
This could be contracted to `ActiveSupport::TimeZone[pref.time_zone || "Etc/UTC"]` like other methods

And as `pref.time_zone` always return something now and falls back to "Etc/UTC", I think the second part `|| ActiveSupport::TimeZone["Etc/UTC"]` is dead code.

So it could simply be `@time_zone ||= ActiveSupport::TimeZone[pref.time_zone]`.

```suggestion
    @time_zone ||= ActiveSupport::TimeZone[pref.time_zone]
```

Reply:
Ah, I got it now, it's in case the user time zone is invalid. It would return `nil`, so it needs to adapt to it. That was the last commit purpose.

Discard my previous comment.